src/services/blocks/BlocksService.ts

```diff
@@ -72,9 +68,11 @@ export class BlocksService extends AbstractService {
 			digest,
 		} = block.header;
 
-		const logs = digest.logs.map((log) => {
-			const { type, index, value } = log;
+		const authorId = sessionValidators
+			? this.extractAuthor(sessionValidators, digest)
+			: undefined;
```

Comment:
What does `extractAuthor` return? If it returns `undefined` this ternary isn't needed.
```suggestion
		const authorId = this.extractAuthor(sessionValidators, digest);
```

Reply:
We just need something so TS does not complain since it knows sessionValidators may be undefined.

Reply:
eew, so you're saying we must do like `const authorId = this.extractAuthor(sessionValidators, digest) || undefined;` just to appease the compiler? :/

Reply:
Yeah, but we need to check before we call `extractAuthor`, because `extractAuthor` requires that the type of the `sessionValidators` param be `Array<AccountId>`. I agree with the compiler here because sending down `undefined` where only an array is expected could lead to issues at runtime.

Reply:
I see what you mean. But is it not an error if `sessionValidators` is `undefined` (or an empty array)? Like, shouldn't we bail before we get here if that's the case?